plugins/connections/ui/src/main/java/org/pentaho/di/vfs/connections/ui/dialog/ConnectionDialog.java

```diff
@@ -80,7 +83,7 @@ public class ConnectionDialog extends Dialog {
 
   Supplier<Spoon> spoonSupplier = Spoon::getInstance;
   private String connectionTypeKey;
-  private ConnectionDetails connectionDetails;
+  private BaseVFSConnectionDetails connectionDetails;
```

Comment:
1. The dialog is used for all connections, even non-VFS ones. Must keep the variable as `ConnectionDetails`.
2. Also, cannot use the optional base class, `BaseVFSConnectionDetails`. Must test if connections are `instanceof VFSConnectionDetails` and only for those, and if they have `isSupportsRootPath() == true` is root path visible.
3. Also, we need to add back `setRootPath` to the interface -- it's clearly needed, after all. Default implementation will do nothing.

Reply:
Addressed.